src/server/job_func.c

```diff
@@ -1545,7 +1545,7 @@ resv_free(resc_resv *presv)
 	 * anywhere. Make it NULL and make sure presv is validated
 	 * before being used.
 	 */
-	presv = (resc_resv *)0;
+	presv = NULL;
```

Comment:
This statement has no effect since it only sets the value in the local stack. Either the caller should set the old pointer to NULL or the function (and callers) be updated to pass resc_resv **presv so we can say *presv = NULL;    The latter suggestion doesn't help much in the case of resv_purge() unless it is updated as well.

Reply:
I've removed the assignment. It seems out of scope to do either change. Also, the caller should know that after it is freed, the pointer is worthless.